src/components/editMeasure/measureDetails/measureInformation/MeasureInformation.test.tsx

```diff
@@ -140,7 +140,90 @@ describe("MeasureInformation component", () => {
   const { getByTestId, queryByText, findByTestId, getByRole, getByText } =
     screen;
 
+  it("Toast error shows when endorsing organization is not null and endorsement Id is empty", async () => {
+    measureStore.state.mockImplementationOnce(() => measure);
+    checkUserCanEdit.mockImplementationOnce(() => true);
+    await act(async () => {
+      render(<MeasureInformation setErrorMessage={setErrorMessage} />);
+      const endorserAutoComplete = await screen.findByTestId("endorser");
+      const endorserId = getByTestId(
+        "endorsement-number-input"
+      ) as HTMLInputElement;
+
+      fireEvent.keyDown(endorserAutoComplete, { key: "ArrowDown" });
+      // selects 2nd option
+      const endorserOptions = await screen.findAllByRole("option");
+      fireEvent.click(endorserOptions[1]);
+
+      // verifies if the option is selected
+      const endorserComboBox =
+        within(endorserAutoComplete).getByRole("combobox");
+      expect(endorserComboBox).toHaveValue("NQF");
+      //verifies endorserId was enabled
+      expect(endorserId).toBeEnabled();
+      //clear endorserId
+      fireEvent.change(endorserId, {
+        target: { value: "" },
+      });
+      expect(endorserId).toHaveValue("");
+
+      const saveButton = await screen.findByRole("button", { name: "Save" });
+      expect(saveButton).toBeInTheDocument();
+      await waitFor(() => expect(saveButton).toBeEnabled());
+      act(() => {
+        fireEvent.click(saveButton);
+      });
+      // previous implementation depended on setTimeout.
+      await waitFor(() => {
+        expect(
+          screen.getByText("Endorser Number is Required")
+        ).toBeInTheDocument();
+      });
+    });
+  });
+
+  it("Toast error shows when endorsement Id is not alphanumeric", async () => {
+    measureStore.state.mockImplementationOnce(() => measure);
+    checkUserCanEdit.mockImplementationOnce(() => true);
+    await act(async () => {
+      render(<MeasureInformation setErrorMessage={setErrorMessage} />);
+      const endorserAutoComplete = await screen.findByTestId("endorser");
+      const endorserId = getByTestId(
+        "endorsement-number-input"
+      ) as HTMLInputElement;
+
+      fireEvent.keyDown(endorserAutoComplete, { key: "ArrowDown" });
+      // selects 2nd option
+      const endorserOptions = await screen.findAllByRole("option");
+      fireEvent.click(endorserOptions[1]);
+
+      // verifies if the option is selected
+      const endorserComboBox =
+        within(endorserAutoComplete).getByRole("combobox");
+      expect(endorserComboBox).toHaveValue("NQF");
```

Comment:
Same suggestion as above: consider asserting on non-null, non-empty string since we'll being add more endorsors in the future.

Reply:
Ignore me, just realized this is merging to the release branch.